apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/rest/endpoint/UtilsController.java

```diff
@@ -0,0 +1,521 @@
+            @FormParam("qrCodeData") String qrCodeData,
+        @Parameter(name = "width", description = "QR code image width, optional") // works on UI without good description
+            @FormParam("width") @DefaultValue("0") String widthStr,
+        @Parameter(name = "height", description = "QR code image height, optional") // works on UI without good description
+            @FormParam("height") @DefaultValue("0") String heightStr
+    ) {
+        log.debug("Started encodeQRCode: \n\t\twidthStr={}, heightStr={}, qrCodeData={}", widthStr, heightStr, qrCodeData);
+        ResponseBuilder response = ResponseBuilder.startTiming();
+        if (StringUtils.isEmpty(qrCodeData)) {
+            return response.error( ApiErrors.MISSING_PARAM, "qrCodeData").build();
+        }
+        int width;
+        try {
+            width = RestParameters.parseInt(widthStr, "width", 0, 5000, false);
+        } catch (RestParameterException e) {
+            return response.error(ApiErrors.INCORRECT_PARAM, "width", widthStr).build();
+        }
+        int height;
+        try {
+            height = RestParameters.parseInt(heightStr, "height", 0, 5000, false);
+        } catch (RestParameterException e) {
+            return response.error(ApiErrors.INCORRECT_PARAM, "height", heightStr).build();
+        }
+
+        QrEncodeDto dto = new QrEncodeDto();
+        try {
+            Map<EncodeHintType, Object> hints = new HashMap<>();
+            // Error correction level: L (7%), M (15%), Q (25%), H (30%) -- Default L.
+            hints.put(EncodeHintType.ERROR_CORRECTION, ErrorCorrectionLevel.M);
+            hints.put(EncodeHintType.MARGIN, 0); // Default 4
+            hints.put(EncodeHintType.CHARACTER_SET, "UTF-8");
+
+            BitMatrix matrix = new MultiFormatWriter().encode(qrCodeData,
+                BarcodeFormat.QR_CODE,
+                width,
+                height,
+                hints
+            );
+            BufferedImage bufferedImage = MatrixToImageWriter.toBufferedImage(matrix);
+            ByteArrayOutputStream os = new ByteArrayOutputStream();
+            ImageIO.write(bufferedImage, "jpeg", os);
+            byte[] bytes = os.toByteArray();
+            os.close();
+            String base64 = Base64.getEncoder().encodeToString(bytes);
+            log.debug("base64 = {}", base64);
+            dto.qrCodeBase64 = base64;
+        } catch(WriterException | IOException | NullPointerException ex) {
+            String errorMessage = String.format("Error creating QR from qrCodeData: %s, ex = %s", qrCodeData, ex.getMessage());
+            log.warn(errorMessage, ex);
+            return response.error(ApiErrors.INTERNAL_SERVER_EXCEPTION, errorMessage).build();
+        }
+        log.debug("encodeQRCode result: {}", dto);
+        return response.bind(dto).build();
+    }
+
+    @Path("/qrcode/decode")
```

Comment:
verb in rest url is prohibited

Reply:
Old api methods looks like RPC methods powered on Rest, so that  no need to follow old RPC-like approach. Nouns in rest url belong to Rest best practices, so that we have to use it whenever possible.

Reply:
Remaned to ....-ing